runtime/taskcontext.go

```diff
@@ -68,13 +76,17 @@ type TaskContextController struct {
 }
 
 // NewTaskContext creates a TaskContext and associated TaskContextController
-func NewTaskContext(tempLogFile string) (*TaskContext, *TaskContextController, error) {
+func NewTaskContext(tempLogFile string, claim *queue.TaskClaimResponse) (*TaskContext, *TaskContextController, error) {
```

Comment:
I don't know why it didn't occur to me, but creating the &taskInfo{} object elsewhere and passing that in might be a lot easier/nicer.